### PR TITLE
LPS-161176 [CP 2.0] Unit Test

### PR DESCRIPTION
**Pull request by joaodamasceno2001**

### CODE REVIEW

**Comment by joaodamasceno2001** [comment]
ci:test:sf

**Review by claricebrasil — CHANGES_REQUESTED** [review]

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/TeamMembers/components/ManageProductUsers/ManageProductUsers.test.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	}),
+}));
+
+const mockedProps = {
+	project: {accountKey: 'Test'},
+	subscriptionGroups: [],
+};
+
+describe('Manage Product Users', () => {
+	beforeEach(() => {
```
[the remark this answers is not in the document]
Comment (joaodamasceno2001) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/TeamMembers/components/ManageProductUsers/ManageProductUsers.test.js:30]: I belive that, if we are gonna use "mockedProps" more than once, we have to use this beforeEach.

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/TeamMembers/components/ManageProductUsers/ManageProductUsers.test.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+		mockedProps.subscriptionGroups = [];
+	});
+
+	it('renders in the screen', () => {
+		mockedProps.subscriptionGroups.push({
+			activationStatus: 'Active',
+			name: 'Liferay Experience Cloud',
+		});
+
+		const {queryByTestId} = render(
```
[the remark this answers is not in the document]
Comment (joaodamasceno2001) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/TeamMembers/components/ManageProductUsers/ManageProductUsers.test.js:40]: I am testing the return of component "MangeProductUsers.js", I created an id and I am using it with  queryByTestId. I think it`s a nice way to find the element using an Id

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/TeamMembers/components/ManageProductUsers/ManageProductUsers.test.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+	});
+
+	it("doesn't renders without valid subscription group", () => {
+		const {queryByTestId} = render(
+			<MockedProvider>
+				<ManageProductUsers {...mockedProps} />
+			</MockedProvider>
+		);
+
+		expect(queryByTestId('test-id')).toBeFalsy();
```
[the remark this answers is not in the document]
Comment (joaodamasceno2001) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/TeamMembers/components/ManageProductUsers/ManageProductUsers.test.js:56]: I thought about that, but if this button only appears when the project is "Active", so I only created a test to this case, but I can created others case too.

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/TeamMembers/components/ManageProductUsers/ManageProductUsers.test.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (claricebrasil) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/TeamMembers/components/ManageProductUsers/ManageProductUsers.test.js:30]: beforeEach is recommended when you want to reset the global data for each block of test. If is not the case, you can just mock the data and pass it as props for the component

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/TeamMembers/components/ManageProductUsers/ManageProductUsers.test.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (claricebrasil) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/TeamMembers/components/ManageProductUsers/ManageProductUsers.test.js:40]: Ok, nice.. you can also find by className or by the text too, if you don't want to add an ID for the element

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/TeamMembers/components/ManageProductUsers/ManageProductUsers.test.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (claricebrasil) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/TeamMembers/components/ManageProductUsers/ManageProductUsers.test.js:56]: Just to make sure every possible case is been tested, I believe would be nice to test with the in-progress status too

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/TeamMembers/components/ManageProductUsers/ManageProductUsers.test.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (joaodamasceno2001) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/TeamMembers/components/ManageProductUsers/ManageProductUsers.test.js:30]: Right, this is what I want, I am reseting the mockedProps for each block of test.

**Comment by joaodamasceno2001** [comment]
ci:test:sf

**Review by claricebrasil — APPROVED** [review]
Nice job, João!

**Review by jairmedeiros — CHANGES_REQUESTED** [review]

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/TeamMembers/components/ManageProductUsers/ManageProductUsers.js [hunk 4]
```diff
@@ -96,7 +96,10 @@ const ManageProductUsers = ({project, subscriptionGroups}) => {
 	return (
 		<>
 			{(isActiveStatusDXPC || isActiveStatusAC || isActiveStatusLXC) && (
-				<div className="bg-brand-primary-lighten-6 border-0 card card-flat cp-manager-product-container mt-5">
+				<div
+					className="bg-brand-primary-lighten-6 border-0 card card-flat cp-manager-product-container mt-5"
+					data-testid="test-id"
```
[the remark this answers is not in the document]
Comment (claricebrasil) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/TeamMembers/components/ManageProductUsers/ManageProductUsers.js:101]: oi Jair, não usamos! Por mais que não seja tão indicado, em últimos casos podemos pegar o elemento pelo nome da classe que nesse caso tem uma classe específica pra ele.

**Pull request closed without merging**